File: scripts/generate_product_pages.rb
#!/usr/bin/env ruby
require 'yaml'
require 'fileutils'

data = YAML.load_file('_data/products.yml')
products = data['products'] || {}
Dir.mkdir('products') unless Dir.exist?('products')

products.each do |slug, info|
  title = info['name']
  desc = info['description'] || ''
  content = <<~YAML
  ---
  layout: product
  title: "#{title}"
  product: #{slug}
  description: "#{desc}"
  og_image: /image/mix1.png
  ---
  YAML

  File.write(File.join('products', "#{slug}.html"), content)
  puts "Wrote products/#{slug}.html"
end

File: scripts/generate_product_pages.py
#!/usr/bin/env python3
import yaml
from pathlib import Path

data = yaml.safe_load(Path(' _data/products.yml'.strip()).read_text())
products = data.get('products', {})
out_dir = Path('products')
out_dir.mkdir(exist_ok=True)

template = '''---
layout: product
title: "{title}"
product: {slug}
description: "{description}"
og_image: /image/mix1.png
---
'''

for slug, info in products.items():
    title = info.get('name')
    description = info.get('description', '')
    content = template.format(title=title.replace('"','\"'), slug=slug, description=description.replace('"','\"'))
    path = out_dir / f"{slug}.html"
    path.write_text(content)
    print('Wrote', path)
